contracts/socket/andromeda-socket-astroport/src/contract.rs

```diff
@@ -0,0 +1,286 @@
+                Some(recipient) => recipient,
+            };
+            swap_and_forward_cw20(
+                ctx,
+                from_asset,
+                Uint128::new(amount.u128()),
+                to_asset,
+                recipient,
+                AndrAddr::from_string(sender),
+                max_spread,
+                minimum_receive,
+                operations,
+            )
+        }
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+fn execute_swap_and_forward(
+    ctx: ExecuteContext,
+    to_asset: Asset,
+    recipient: Option<Recipient>,
+    max_spread: Option<Decimal>,
+    minimum_receive: Option<Uint128>,
+    operations: Option<Vec<SwapOperation>>,
+) -> Result<Response, ContractError> {
+    let fund = one_coin(&ctx.info).map_err(|_| ContractError::InvalidAsset {
+        asset: "Invalid or missing coin".to_string(),
+    })?;
+
+    let from_asset = Asset::NativeToken(fund.denom);
+    let sender = AndrAddr::from_string(&ctx.info.sender);
+    let recipient = match recipient {
+        None => Recipient::new(sender.clone(), None),
+        Some(recipient) => recipient,
+    };
+
+    let swap_msg = execute_swap_astroport_msg(
+        ctx,
+        from_asset.clone(),
+        fund.amount,
+        to_asset.clone(),
+        recipient.clone(),
+        sender,
+        max_spread,
+        minimum_receive,
+        operations,
+    )?;
+
+    Ok(Response::default()
+        .add_submessage(swap_msg)
+        .add_attributes(vec![
+            attr("from_asset", from_asset.to_string()),
+            attr("from_amount", fund.amount),
+            attr("to_asset", to_asset.to_string()),
+            attr("recipient", recipient.get_addr()),
+        ]))
+}
+
+#[allow(clippy::too_many_arguments)]
+fn swap_and_forward_cw20(
+    ctx: ExecuteContext,
+    from_asset: Asset,
+    from_amount: Uint128,
+    to_asset: Asset,
+    recipient: Recipient,
+    refund_addr: AndrAddr,
+    max_spread: Option<Decimal>,
+    minimum_receive: Option<Uint128>,
+    operations: Option<Vec<SwapOperation>>,
+) -> Result<Response, ContractError> {
+    let swap_msg = execute_swap_astroport_msg(
+        ctx,
+        from_asset.clone(),
+        from_amount,
+        to_asset.clone(),
+        recipient.clone(),
+        refund_addr,
+        max_spread,
+        minimum_receive,
+        operations,
+    )?;
+
+    Ok(Response::default()
+        .add_submessage(swap_msg)
+        .add_attributes(vec![
+            attr("from_asset", from_asset.to_string()),
+            attr("from_amount", from_amount),
+            attr("to_asset", to_asset.to_string()),
+            attr("recipient", recipient.get_addr()),
+        ]))
+}
+
+fn execute_update_swap_router(
+    ctx: ExecuteContext,
+    swap_router: AndrAddr,
+) -> Result<Response, ContractError> {
+    let sender = ctx.info.sender;
+    ensure!(
+        ADOContract::default().is_owner_or_operator(ctx.deps.storage, sender.as_ref())?,
```

Comment:
`ADOContract::default()` can be replaced with `ctx.contract`

Reply:
Also, it's better to achieve this validation through our attributes on the `ExecuteMsg` level. This msg would use `restricted`